miden-lib/build.rs

```diff
@@ -225,12 +239,16 @@ fn parse_proc_offsets(filename: impl AsRef<Path>) -> Result<BTreeMap<String, usi
 fn compile_miden_lib(
     source_dir: &Path,
     target_dir: &Path,
-    assembler: Assembler,
+    mut assembler: Assembler,
 ) -> Result<Library> {
     let source_dir = source_dir.join(ASM_MIDEN_DIR);
+    let shared_path = Path::new(ASM_DIR).join(SHARED_DIR);
+
+    let miden_namespace = "miden".parse::<LibraryNamespace>().expect("invalid base namespace");
+    // add the shared modules to the kernel lib under the miden::util namespace
+    assembler.add_modules_from_dir(miden_namespace.clone(), &shared_path)?;
 
-    let namespace = "miden".parse::<LibraryNamespace>().expect("invalid base namespace");
-    let miden_lib = Library::from_dir(source_dir, namespace, assembler)?;
+    let miden_lib = Library::from_dir(source_dir, miden_namespace, assembler)?;
```

Comment:
To confirm, this does put include procedures from the shared modules into `miden_lib` library, right?

Reply:
This a bit tricky now, unfortunately.

The modules in miden lib can use `miden::util::*` but users importing miden lib won't be able to.

This is because with `Library::from_dir` we're only including the modules from `miden-lib/asm/miden` in the library, but not the ones from `miden-lib/asm/shared`. The latter ones are only available during assembly.

This is why it's necessary to reexport all of them from the kernel and miden lib modules so users still can access these procedures.

I hope that answers your question.